### PR TITLE
fix: Use single policy statement

### PR DESCRIPTION
**Pull request by taion**

### CODE REVIEW

**Comment by taion** [comment]
@sloria @emmilco adding new poll handlers is broken without this

File: tqp/topic_queue_poller.py [hunk 1]
```diff
@@ -219,26 +219,31 @@ def ensure_queue(self):
...
+                    "Policy": {
+                        "Version": "2012-10-17",
+                        "Statement": {
+                            "Sid": "sns",
+                            "Effect": "Allow",
+                            "Principal": {"AWS": "*"},
+                            "Action": "SQS:SendMessage",
+                            "Resource": queue_arn,
+                            "Condition": {
+                                "ArnEquals": {"aws:SourceArn": topic_arns}
```
Comment (emmilco) [inline tqp/topic_queue_poller.py:242]: Are you sure this works? See here: https://docs.aws.amazon.com/sns/latest/dg/sns-access-policy-language-api-permissions-reference.html

`Each policy must cover only a single topic (when writing a policy, don't include statements that cover different topics)`

Comment (emmilco) [reply tqp/topic_queue_poller.py:242]: (It's possible that that's not relevant, since this is an SQS policy.)

Comment (itajaja) [reply tqp/topic_queue_poller.py:242]: Here's what I get from the UI:

image

maybe you need

```suggestion
                                "ArnEquals": {"aws:SourceArn": ",".join(topic_arns)}
```

? just double check either way

Comment (sloria) [reply tqp/topic_queue_poller.py:242]: @emmilco I think that specific docco is only applicable to SNS.

That said, I also can't find any docs suggesting that an array will work here.

From https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_condition_operators.html#Conditions_ARN:

> Case-sensitive matching of the ARN. Each of the six colon-delimited components of the ARN is checked separately and each can include a multi-character match wildcard (*) or a single-character match wildcard (?). These behave identically.

Can we do wildcard matching here instead?

**Comment by taion** [comment]
seems to work

**Comment by taion** [comment]
on the array thing, i'm following https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_multi-value-conditions.html

**Pull request merged**